runtime/smithy-client/common/src/aws/smithy/kotlin/runtime/client/SdkClientConfig.kt

```diff
@@ -41,6 +46,11 @@ public interface SdkClientConfig {
      * @param TConfig the type of configuration built by this builder
      */
     public interface Builder<TConfig : SdkClientConfig> : Buildable<TConfig> {
+        /**
+         * A reader-friendly name for the client.
+         */
+        public var clientName: String
+
```

Comment:
**Suggestion**: We should be clearer here and elsewhere about how the client name will be used, why someone might want to set it, and what the default is.